LoopKit/GlucoseKit/GlucoseStore.swift

```diff
@@ -373,15 +376,17 @@ extension GlucoseStore {
         // "eligible" for storing in HealthKit (i.e. are not too "recent" according to `healthKitStorageDelay`
         cacheStore.managedObjectContext.performAndWait {
             let request: NSFetchRequest<CachedGlucoseObject> = CachedGlucoseObject.fetchRequest()
-            let now = Date()
-            let notInHealthKitPredicate = NSPredicate(format: "uuid = nil")
-            let eligibleForStorageInHealthKit = NSPredicate(format: "startDate <= %@", now.addingTimeInterval(-healthKitStorageDelay) as NSDate)
-            request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [notInHealthKitPredicate, eligibleForStorageInHealthKit])
+            request.predicate = NSPredicate(format: "healthKitEligibleDate != nil AND healthKitEligibleDate <= %@", Date() as NSDate)
             do {
                 let stored = try self.cacheStore.managedObjectContext.fetch(request)
                 guard !stored.isEmpty else {
                     return
                 }
+                
+                if stored.contains(where: { $0.uuid != nil }) {
+                    self.log.error("Invalid data! Found CachedGlucoseObjects with non-nil uuid. Should never happen!")
+                    // Note the UUIDs will be overwritten below...hopefully that's ok?
```

Comment:
Actually, thinking through this scenario...

First, it should never happen.

Second, even if it does, HealthKit should recover fine because the old and new HealthKit object will share the same `syncIdentifier`. Therefore, HealthKit will "replace" the old object (with old `uuid`) with the new object (with new `uuid`). There may be an unnecessary round-trip to synchronize the HealthKit objects in DIY (due to a `syncVersion` that doesn't match, but it shouldn't cause an actual issue.

Reply:
In fact, a similar scenario already existed.  If the HealthKit write succeeds, but the subsequent CoreData write fails (to set the `uuid` and clear the `healthKitEligibleDate`). The store will write a "new" object to HealthKit (same data with different `uuid`) and do the same "replace" since the `syncIdentifiers` match.

Reply:
Ah, good point.  Did you want me to remove this comment then?

Reply:
I think updating the error and comment to indicate that it is an unexpected state, but that HealthKit should resolve the issue, would be best.